test/Microsoft.Dnx.ApplicationHost.FunctionalTests/DllImportTests.cs

```diff
@@ -0,0 +1,44 @@
+﻿using Microsoft.Extensions.PlatformAbstractions;
+using Microsoft.Dnx.Testing.Framework;
+using Xunit;
+
+namespace Microsoft.Dnx.ApplicationHost.FunctionalTests
+{
+    public class DllImportTests : DnxSdkFunctionalTestBase
+    {
+        [Theory, TraceTest]
+        [MemberData(nameof(DnxSdks))]
+        public void VerifyDllImportWithProjectReference(DnxSdk sdk)
+        {
+            VerifyDllImport(sdk, "ProjectReferenceTest");
+        }
+
+        [Theory, TraceTest]
+        [MemberData(nameof(DnxSdks))]
+        public void VerifyDllImportWithPackageReference(DnxSdk sdk)
+        {
+            VerifyDllImport(sdk, "PackageReferenceTest");
+        }
+
+        public void VerifyDllImport(DnxSdk sdk, string projectName)
+        {
+            var runtimeEnvironment = PlatformServices.Default.Runtime;
+
+            // Preloading .so's on Linux does not work and setting
+            // LD_LIBRARY_PATH is not an option so skipping
+            if (runtimeEnvironment.OperatingSystem == "Linux" && runtimeEnvironment.RuntimeType == "Mono")
```

Comment:
I thought it was not possible to skip for only Mono on Linux. If it is I will fix.